src/Microsoft.AspNetCore.Razor.Language/Intermediate/HtmlAttributeValueIRNode.cs

```diff
@@ -10,16 +10,14 @@ public sealed class HtmlAttributeValueIRNode : RazorIRNode
     {
         public override ItemCollection Annotations => ReadOnlyItemCollection.Empty;
 
-        public override IList<RazorIRNode> Children { get; } = EmptyArray;
+        public override IList<RazorIRNode> Children { get; } = new List<RazorIRNode>();
```

Comment:
For these attribute value IR nodes we can initialize their children to a capacity of 1 since that's the default behavior for all these guys. Not sure if that's already the default for List but food for thought.